addons/mail/models/discuss/discuss_channel.py

```diff
@@ -673,6 +673,80 @@ def _add_members(
                     current_channel_member.sudo()._rtc_invite_members(member_ids=new_members.ids)
         return all_new_members
 
+    def invite_by_email(self, addresses):
+        """
+        Send channel invitation emails to a list of email addresses.
+
+        - Existing members' email addresses are ignored.
+        - An invitation can only be re-sent to the same email address for the
+        same channel after 5 minutes.
+
+        :param addresses: List of email addresses to invite.
+        :type addresses: list[str]
+
+        """
+        self.check_access("read")
+        if not self._allow_invite_by_email():
+            raise UserError(
+                self.env._("Inviting by email is not allowed for this channel type (%s).")
+                % self.channel_type
+            )
+        addresses = {
+            norm for addr in addresses if addr and (norm := email_normalize(addr))
+        }
+        # Removing already sent email in the past 5 minutes to avoid spamming. However, it's
+        # sometimes acceptable to re-send an invitation (e.g. the recipient deleted the email).
+        #
+        # sudo - mail.mail: accessing mail records to determine which emails were already sent
+        # is acceptable.
+        addresses -= set(
```

Comment:
Instead of that custom stuff based on references and email_to, you may set model / res_id and check mail on that specific channel, and fetch all TOs, filter them out from the input list. Will be more robust I think.